packages/caver-wallet/src/keyring/signatureData.js

```diff
@@ -0,0 +1,118 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const utils = require('../../../caver-utils')
+
+/**
+ * Representing a SignatureData class that includes ECDSA signature data string.
+ * @class
+ */
+class SignatureData {
+    /**
+     * creates a privateKey.
+     * @param {string} key - The private key string.
+     */
+    constructor(signature) {
+        if (signature instanceof SignatureData) {
+            this.v = signature.v
+            this.r = signature.r
+            this.s = signature.s
+        }
+        if (!_.isArray(signature)) signature = utils.resolveSignature(signature)
+
+        const [v, r, s] = signature
+        this.v = v
+        this.r = r
+        this.s = s
+    }
+
+    /**
+     * @type {string}
+     */
+    get v() {
+        return this._v
+    }
+
+    set v(v) {
+        this._v = utils.makeEven(v)
+    }
+
+    get V() {
+        return this.v
+    }
+
+    set V(v) {
+        this.v = v
+    }
+
+    /**
+     * @type {string}
+     */
+    get r() {
+        return this._r
+    }
+
+    set r(r) {
+        this._r = utils.makeEven(r)
+    }
+
+    get R() {
+        return this.r
+    }
+
+    set R(r) {
+        this.r = r
+    }
+
+    /**
+     * @type {string}
+     */
+    get s() {
+        return this._s
+    }
+
+    set s(s) {
+        this._s = utils.makeEven(s)
+    }
+
+    get S() {
+        return this.s
+    }
+
+    set S(s) {
+        this.s = s
+    }
+
+    isEmpty() {
+        if (this.v === '0x01' && this.r === '0x' && this.s === '0x') return true
+        return false
+    }
+
+    encode() {
+        return [this.v, this.r, this.s]
+    }
+
+    toString() {
```

Comment:
Don't we need to add function description?

Reply:
I added descriptions!